### DISCUSSION TITLE
What's the point of `config = function() end`?

### DISCUSSION THREAD

**Discussion by rieje** [LazyVim/LazyVim#5023] [Q&A] [answered]
What's the point of `config = function() end` [here](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/lsp/init.lua#L8) (empty function)? If the point is to not load the setup function, won't omitting both 'opts' and 'config' achieving the same effect?

I'm looking at examples on how to use lazy.nvim to manage related plugins--at first I was thinking of nested dependencies and trying to put everything that belongs into a plugin into its own plugin spec, but now it seems it's simpler to simply declare the dependencies and do all the setup and glue in the parent plugin instead when possible.

Any advice much appreciated.

**Comment by folke** [comment, marked as answer]
If a user would set opts for that plugin, it's setup function would be called.
By adding an empty `config` it won't do anything.
That plugin is setup inside the lspconfig setup
